File: Android/app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/filter/BottomSheetChip.java
package com.lucagiorgetti.surprix.ui.mainfragments.filter;

import android.content.Context;
import android.util.AttributeSet;

import com.google.android.material.chip.Chip;

class BottomSheetChip extends Chip {
    public BottomSheetChip(Context context, ChipFilter chipFilter) {
        super(context);
        this.setText(chipFilter.getName());
        this.setCloseIconVisible(true);
        this.setCheckedIconVisible(true);
        this.setCloseIconVisible(false);
        this.setCheckable(true);
        this.setChecked(chipFilter.isSelected());
    }

    public BottomSheetChip(Context context) {
        super(context);
    }

    public BottomSheetChip(Context context, AttributeSet attrs) {
        super(context, attrs);
    }
}

File: app/src/main/java/com/lucagiorgetti/surprix/ui/mainfragments/filter/BottomSheetChip.kt
package com.lucagiorgetti.surprix.ui.mainfragments.filter

import android.content.Context
import android.util.AttributeSet
import com.google.android.material.chip.Chip

internal class BottomSheetChip : Chip {
    constructor(context: Context?, chipFilter: ChipFilter?) : super(context) {
        this.text = chipFilter?.name
        this.isCloseIconVisible = true
        this.isCheckedIconVisible = true
        this.isCloseIconVisible = false
        this.isCheckable = true
        this.isChecked = chipFilter!!.isSelected
    }

    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
}
